marc_to_solr/lib/princeton_marc.rb

```diff
@@ -341,20 +343,208 @@ def oclc_normalize oclc, opts = {prefix: false}
   end
 end
 
+# Cached mapping of ARKs to Bib IDs
+# Retrieves and stores paginated Solr responses containing the ARK's and BibID's
+class CacheMap
+  attr_reader :values
+
+  # Constructor
+  # @param endpoint [String] the URL endpoint for the Solr
+  # @param rows [Integer] the number of rows for each Solr response
+  # @param logger [IO] the logging device
+  def initialize(endpoint:, rows: 1000000, logger: STDOUT)
+    @endpoint = endpoint
+    @rows = rows
+    @logger = logger
+    @values = {}
+
+    seed!
+  end
+
+  # Seed the cache
+  # @param page [Integer] the page number at which to start the caching
+  def seed!(page: 1)
+    response = query(page: page)
+    return if response.empty?
+
+    pages = response.fetch('pages')
+
+    cache_page(response)
+
+    if pages.fetch('last_page?') == false
+      seed!(page: page + 1)
+    end
+  end
+
+  private
+
+    # Cache a page
+    # @param page [Hash] Solr response page
+    def cache_page(page)
+      docs = page.fetch('docs')
+      docs.each do |doc|
+        arks = doc.fetch('identifier_ssim', [])
+        bib_ids = doc.fetch('source_metadata_identifier_ssim', [])
+
+        ark = arks.first
+        bib_id = bib_ids.first
+
+        @values[ark] = bib_id
+      end
+    end
+
+    # Query the service using the endpoint
+    # @param [Integer] the page parameter for the query
+    def query(page: 1)
+      begin
+        http_response = Faraday.get("#{@endpoint}.json&q=&rows=#{@rows}&page=#{page}")
```

Comment:
Do we want to limit to records that have ARKs?  I think `f[identifier_tesim][]=ark` would do it.

Reply:
Thank you for identifying this, the query has been updated accordingly